plugins/optimization-detective/storage/class-od-storage-lock.php

```diff
@@ -20,6 +20,49 @@
  */
 final class OD_Storage_Lock {
 
+	/**
+	 * Capability for being able to store a URL Metric now.
+	 *
+	 * @since n.e.x.t
+	 * @access private
+	 * @var string
+	 */
+	const STORE_URL_METRIC_NOW_CAPABILITY = 'od_store_url_metric_now';
+
+	/**
+	 * Adds hooks.
+	 *
+	 * @since n.e.x.t
+	 * @access private
+	 */
+	public static function add_hooks(): void {
+		add_filter( 'map_meta_cap', array( __CLASS__, 'filter_map_meta_cap' ), 10, 3 );
+	}
+
+	/**
+	 * Filters map_meta_cap to grant the `od_store_url_metric_now` capability to administrators by default.
+	 *
+	 * @since n.e.x.t
+	 * @access private
+	 *
+	 * @param string[]|mixed $caps    Primitive capabilities required of the user.
+	 * @param string         $cap     Capability being checked.
+	 * @param int            $user_id The user ID.
+	 * @return string[] Primitive capabilities required of the user.
+	 */
+	public static function filter_map_meta_cap( $caps, string $cap, int $user_id ): array {
+		if ( ! is_array( $caps ) ) {
+			$caps = array();
+		}
+
+		$primitive_cap = 'manage_options';
+		if ( self::STORE_URL_METRIC_NOW_CAPABILITY === $cap && user_can( $user_id, $primitive_cap ) ) {
```

Comment:
The second check here is unnecessary, as that is checked after running the `map_meta_cap` filter anyway.
```suggestion
		if ( self::STORE_URL_METRIC_NOW_CAPABILITY === $cap ) {
```

Any `map_meta_cap` callback should return the list of capabilities required, and then Core will check that the user has all of them. So we don't need to check the same thing here too prior to that.

We can then also get rid of the `$user_id` parameter that is no longer used.

Reply:
Good point. Done in bbba00163b8d949b00c9553484dcd606757cf62e.